src/python/python-proto/python_proto/metadata.py

```diff
@@ -0,0 +1,2 @@
+# N tuples of (str, str)
+GrpcOutboundMetadata = tuple[tuple[str, str | bytes], ...]
```

Comment:
Is there a reason this is an unbounded tuple of tuples? That seems like a more awkward data type to interact with than a dict.

Reply:
this is unfortunately what grpc expects at this callsite:
```
    def query_with_uid(
        self,
        node_uid: Uid,
        graph_query: GraphQuery,
        metadata: GrpcOutboundMetadata | None = None,
    ) -> QueryGraphWithUidResponse:
        request = QueryGraphWithUidRequest(
            node_uid=node_uid,
            graph_query=graph_query,
        )
        proto_response = self.proto_client.QueryGraphWithUid(
            request.into_proto(),
            timeout=5,
            metadata=metadata,  # <------- here
```

i'm happy to keep it wrapped as a dict up to that point?

Reply:
At first glance, that would seem to be the way to go; keep all the GRPC nastiness confined to as small a footprint as possible and make the rest of the code easier to interact with.